Index: src/modules/engines/gl_sdl/evas_engine.c
===================================================================
--- src/modules/engines/gl_sdl/evas_engine.c	(revision 50138)
+++ src/modules/engines/gl_sdl/evas_engine.c	(working copy)
@@ -97,7 +97,7 @@
 	  }
      }
 
-   evas_gl_common_context_resize(re->gl_context, w, h);
+   evas_gl_common_context_resize(re->gl_context, w, h,re->gl_context->rot);
 }
 
 static void
@@ -114,7 +114,7 @@
    Render_Engine *re;
 
    re = (Render_Engine *)data;
-   evas_gl_common_context_resize(re->gl_context, re->w, re->h);
+   evas_gl_common_context_resize(re->gl_context, re->w, re->h,re->gl_context->rot);
    /* smple bounding box */
    if (!re->draw.redraw)
      {
@@ -1410,7 +1410,7 @@
 	return NULL;
      }
    evas_gl_common_context_use(re->gl_context);
-   evas_gl_common_context_resize(re->gl_context, w, h);
+   evas_gl_common_context_resize(re->gl_context, w, h,re->gl_context->rot);
 
    /* End GL Initialization */
    re->w = w;
