diff -u fontconfig-2.3.2/configure.in fontconfig-2.3.2-new/configure.in
--- fontconfig-2.3.2/configure.in	2005-04-27 13:54:24.000000000 -0500
+++ fontconfig-2.3.2-new/configure.in	2010-01-12 02:54:22.000000000 -0600
@@ -267,18 +267,10 @@
 		expat=no
 		;;
 	yes)
-		AC_CHECK_FUNCS(XML_SetDoctypeDeclHandler)
-		case "$ac_cv_func_XML_SetDoctypeDeclHandler" in
-		yes)
-			HAVE_EXPAT=1
-			AC_SUBST(HAVE_EXPAT)
-			AC_DEFINE_UNQUOTED(HAVE_EXPAT,$HAVE_EXPAT,
-			[Found a useable expat library])
-			;;
-		*)
-			expat=no
-			;;
-		esac
+		HAVE_EXPAT=1
+		AC_SUBST(HAVE_EXPAT)
+		AC_DEFINE_UNQUOTED(HAVE_EXPAT,$HAVE_EXPAT,
+		[Found a useable expat library])
 		;;
 	esac
 	CPPFLAGS="$expatsaved_CPPFLAGS"
diff -u fontconfig-2.3.2/fontconfig.pc.in fontconfig-2.3.2-new/fontconfig.pc.in
--- fontconfig-2.3.2/fontconfig.pc.in	2003-03-05 00:09:14.000000000 -0600
+++ fontconfig-2.3.2-new/fontconfig.pc.in	2010-01-12 15:50:40.000000000 -0600
@@ -6,5 +6,5 @@
 Name: Fontconfig
 Description: Font configuration and customization library
 Version: @VERSION@
-Libs: -L${libdir} -lfontconfig
+Libs: -L${libdir} -Wl,-rpath,${libdir} -lfontconfig
 Cflags: -I${includedir}
