diff --git a/navit/navit/graphics/sdl/Makefile.am b/navit/navit/graphics/sdl/Makefile.am
index 084bf3e..af57ff3 100644
--- a/navit/navit/graphics/sdl/Makefile.am
+++ b/navit/navit/graphics/sdl/Makefile.am
@@ -2,4 +2,4 @@ include $(top_srcdir)/Makefile.inc
 AM_CPPFLAGS = @NAVIT_CFLAGS@ @SDL_CFLAGS@ @FREETYPE2_CFLAGS@ @FONTCONFIG_CFLAGS@ -I$(top_srcdir)/navit -DMODULE=graphics_sdl
 modulegraphics_LTLIBRARIES = libgraphics_sdl.la
 libgraphics_sdl_la_SOURCES = graphics_sdl.c raster.c raster.h
-libgraphics_sdl_la_LDFLAGS = @SDL_LIBS@ @SDL_IMAGE_LIBS@ @FREETYPE2_LIBS@ @FONTCONFIG_LIBS@ -module -avoid-version
+libgraphics_sdl_la_LDFLAGS = @SDL_LIBS@ @SDL_IMAGE_LIBS@ @FREETYPE2_LIBS@ @FONTCONFIG_LIBS@ -lpdl -module -avoid-version
